src/spyglass/decoding/v1/pipeline_waveform_feature_extraction.py

```diff
@@ -0,0 +1,219 @@
+    # Assume SpikeSortingOutput merge entry 'my_sorting_merge_id' exists
+    # Assume 'amplitude' params exist in WaveformFeaturesParams
+    # Assume 'run_interval' exists in IntervalList for the session
+
+    merge_id = 'replace_with_actual_spikesorting_merge_id' # Placeholder
+    feat_params = 'amplitude'
+    interval = 'run_interval'
+
+    # --- Run Waveform Feature Extraction for all spikes ---
+    populate_spyglass_waveform_features_v1(
+        spikesorting_merge_ids=merge_id,
+        features_param_name=feat_params,
+        display_progress=True
+    )
+
+    # --- Run Waveform Feature Extraction restricted to an interval ---
+    # populate_spyglass_waveform_features_v1(
+    #     spikesorting_merge_ids=merge_id,
+    #     features_param_name=feat_params,
+    #     interval_list_name=interval,
+    #     display_progress=True
+    # )
+
+    # --- Run for multiple merge IDs ---
+    # merge_ids = ['id1', 'id2']
+    # populate_spyglass_waveform_features_v1(
+    #     spikesorting_merge_ids=merge_ids,
+    #     features_param_name=feat_params,
+    #     display_progress=True
+    # )
+    ```
+    """
+
+    # --- Input Validation ---
+    params_key = {"features_param_name": features_param_name}
+    if not (WaveformFeaturesParams & params_key):
+        raise ValueError(
+            f"WaveformFeaturesParams not found: {features_param_name}"
+        )
+
+    if isinstance(spikesorting_merge_ids, str):
+        spikesorting_merge_ids = [spikesorting_merge_ids]
+    if not isinstance(spikesorting_merge_ids, list):
+        raise TypeError("spikesorting_merge_ids must be a string or list.")
+
+    valid_merge_ids = []
+    nwb_file_name = None  # To check interval list
+    for merge_id in spikesorting_merge_ids:
+        ss_key = {"merge_id": merge_id}
+        if not (SpikeSortingOutput & ss_key):
+            raise ValueError(f"SpikeSortingOutput entry not found: {merge_id}")
+        valid_merge_ids.append(merge_id)
+        # Get nwb_file_name from the first valid merge_id to check interval
+        if nwb_file_name is None:
+            try:
+                nwb_file_name = (
+                    SpikeSortingOutput.merge_get_parent(ss_key) & ss_key
+                ).fetch1("nwb_file_name")
+            except Exception as e:
+                logger.warning(
+                    f"Could not fetch nwb_file_name for merge_id {merge_id}: {e}"
+                )
+                # Continue, interval check might fail later if interval_list_name is provided
+
+    if interval_list_name and nwb_file_name:
+        interval_key = {
+            "nwb_file_name": nwb_file_name,
+            "interval_list_name": interval_list_name,
+        }
+        if not (IntervalList & interval_key):
+            raise ValueError(
+                f"IntervalList not found: {nwb_file_name}, {interval_list_name}"
+            )
+    elif interval_list_name and not nwb_file_name:
```

Comment:
Isn't `interval_list_name` a primary key of `IntervalList`? if they only passed the interval name, you can use that to retrieve the `nwb_file_name`